ClosedTypeHierarchyDiagnosticSuppressor.Tests/SwitchStatementSuppressorTests.cs

```diff
@@ -94,6 +94,28 @@ public static void DoSwitch(Root root)
         return EnsureNotSuppressed(code, NullableContextOptions.Disable);
     }
 
+    [Test]
+    public Task When_record_with_destructure_And_all_subtypes_match_Then_suppress()
+    {
+        var code = CodeHelper.WrapInNamespace(TypeHierarchies.ProtectedCopyConstructorOnly.PositionalParameter + @"
+static class SwitchTest
+{
+    public static void DoSwitch(Root root)
+    {
+        switch(root)
+        {
+            case Root.Leaf1(var v):
```

Comment:
please add a negative test where e. g. a `Leaf1(object Value)` is matched with `Leaf1(string Value)` and assert that the exhaustiveness warning is not suppressed